lldb/include/lldb/Utility/ProcessInfo.h

```diff
@@ -147,96 +148,111 @@ class ProcessInstanceInfo : public ProcessInfo {
   ProcessInstanceInfo() = default;
 
   ProcessInstanceInfo(const char *name, const ArchSpec &arch, lldb::pid_t pid)
-      : ProcessInfo(name, arch, pid), m_euid(UINT32_MAX), m_egid(UINT32_MAX),
-        m_parent_pid(LLDB_INVALID_PROCESS_ID) {}
+      : ProcessInfo(name, arch, pid) {}
 
   void Clear() {
     ProcessInfo::Clear();
-    m_euid = UINT32_MAX;
-    m_egid = UINT32_MAX;
-    m_parent_pid = LLDB_INVALID_PROCESS_ID;
+    m_euid = std::nullopt;
+    m_egid = std::nullopt;
+    m_parent_pid = std::nullopt;
   }
 
-  uint32_t GetEffectiveUserID() const { return m_euid; }
+  uint32_t GetEffectiveUserID() const { return m_euid.value(); }
```

Comment:
can we actually return `std::optional<uint32_t>` here? Otherwise if someone calls this and `m_euid` doesn't have a value, then it will throw a `std::bad_optional_access` exception.

There is a `std::optional<T>::value_or(T)`, but that will defeat the reason for adding a std::optional in the first place. So I see two options:
```
uint32_t GetEffectiveUserID(uint32_t default_value = 0) const { return m_euid.value_or(default_value); }
```
or
```
std::optional<uint32_t> GetEffectiveUserID() const { return m_euid; }
```
I would prefer the latter, but now sure now many locations this will affect.

Reply:
I don't have particularly strong feelings here but we should keep it consistent between this class and the parent class which has GetUID and GetGID neither of which returns a std::optional. Given the risk of std::bad_optional_access getting thrown thoughout I am coming to the notion that maybe we shouldn't use optionals here? If we have optionals I guess we are mandating that the paired IsValid must exist and must always be called. That seems like it'll create challenges. Can we go back to using invalid values like UINT32_MAX and maybe alter the interface to use optionals in a subsequent pr? This way we could debate the merits of this approach independent of adding a niceness field.

Reply:
Maybe a better approach is to keep the optionals in the members that are particular to ProcessInstanceInfo (i.e. the time fields, priority, and the zombie fields) and revert the pid fields and the pid and uid/gid fields to the previous and punt that to a later pr. How does that sound?

Reply:
@clayborg I made the changes referenced above (i.e. backed out the optionals for the pid/uid/gid fields). Is this a dealbreaker for you? Or do you think we can revisit at a later time?

Reply:
It is ok to not do all of the ivars at one, but for the `std::optional` stuff that remains, if we just call `.value()` and it hasn't been set we are going to throw an exception right? We either need to make these not optional, or return the optional when accessing and let the callers deal with the values no being available.

Reply:
yeah. I of the same view